hoomd/hpmc/pytest/test_shape.py

```diff
@@ -2,46 +2,44 @@
 import hoomd.hpmc
 from hoomd.hpmc import _hpmc
 import numpy as np
+import pytest
 
 
 def test_dict_conversion(shape_dict_conversion_args):
     for shape_params, args_list in shape_dict_conversion_args():
         for args in args_list:
             test_shape = shape_params(args)
             test_dict = test_shape.asDict()
-            for key, val in list(args.items()):
-                if isinstance(test_dict[key], list) and len(test_dict[key]) > 0 \
-                and key != 'shapes':
+            for key, val in args.items():
```

Comment:
The logic for checking all items of a dictionary being similar is the same here and in `test_shape_attached`. You could potentially write a non test function with the logic and just call it here and in the other function appropriately.